analyses/mutational-signatures/03-de_novo_range_of_nsignatures.sh

```diff
@@ -13,31 +13,45 @@ cd "$(dirname "${BASH_SOURCE[0]}")"
 QUICK_MUTSIGS=${QUICK_MUTSIGS:-0}
 
 scratch_dir=../../scratch/mutational-signatures
-denovo_plot_dir=plots/denovo
+denovo_plot_dir=plots/denovo/gof
 
-# if abbreviated mutsigs is "true", we'll run only one number of signatures and
-# for an unacceptably low number of iterations to do any analysis with
+# Directory to hold the goodness-of-fit plot
+mkdir -p $denovo_plot_dir
+
+# The MAF file we'll use is going to WGS samples only
+maf_file=${scratch_dir}/pbta-snv-consensus-wgs.tsv.gz
+
+# If abbreviated mutsigs is "true", we'll extract 3 signatures using
+# an unacceptably low number of iterations to do any analysis with
 if [ "$QUICK_MUTSIGS" -gt "0" ]
 then
-  FLOOR=10
-  CEILING=10
-  NUM_ITER=10
+  # De novo signatures extraction
+  Rscript --vanilla \
+    scripts/de_novo_signature_extraction.R \
+    --maf_file ${maf_file} \
+    --nsignatures_floor 3 \
+    --nsignatures_ceiling 3 \
+    --num_iterations 10 \
+    --seed 42 
 else
-  FLOOR=5
-  CEILING=15
-  NUM_ITER=1000
-fi
+  for model in multinomial poisson; do
+    for seed in {1..5}; do
+      for nsignatures in {2..8}; do
+
+         # De novo signatures extraction
+         Rscript --vanilla \
+           scripts/de_novo_signature_extraction.R \
+           --maf_file ${maf_file} \
+           --nsignatures_floor ${nsignatures} \
+           --nsignatures_ceiling ${nsignatures} \
+           --num_iterations 1000 \
+           --model ${model}
+           --seed ${seed} \
+           --plot_output "${denovo_plot_dir}/gof_seed_${seed}_model_${model}.pdf"
+      
+      done
```

Comment:
Good point about the plots, we could do:

```suggestion

       # De novo signatures extraction
       Rscript --vanilla \
         scripts/de_novo_signature_extraction.R \
         --maf_file ${maf_file} \
         --nsignatures_floor 2 \
         --nsignatures_ceiling 8 \
         --num_iterations 1000 \
         --model ${model}
         --seed ${seed} \
         --plot_output "${denovo_plot_dir}/gof_seed_${seed}_model_${model}.pdf"
      

```

And I think that should have the same memory considerations as what's in https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/974, right?

Reply:
Ah wait I missed this - definitely no 2:8 loop since the range _is_ the `{sigfit}` argument. 

This is OK memory-wise. We just don't really want to go above k=10, unless we reduce iterations to 500 or so, which would be a step in the wrong direction.